inc/Ui/Revisions.php

```diff
@@ -0,0 +1,437 @@
+
+    /** 
+     * Revisions Ui constructor
+     *
+     * @param int $first  skip the first n changelog lines
+     * @param bool|string $media_id  id of media, or false for current page
+     */
+    public function __construct($first = 0, $media_id = false)
+    {
+        $this->first    = $first;
+        $this->media_id = $media_id;
+    }
+
+    /**
+     * Display list of old revisions
+     *
+     * @author Andreas Gohr <andi@splitbrain.org>
+     * @author Ben Coburn <btcoburn@silicodon.net>
+     * @author Kate Arzamastseva <pshns@ukr.net>
+     * @author Satoshi Sahara <sahara.satoshi@gmail.com>
+     *
+     * @return void
+     */
+    public function show()
+    {
+        global $ID;
+
+        if ($this->media_id) {
+            return $this->showMediaRevisions($this->media_id);
+        } else {
+            return $this->showPageRevisions($ID);
+        }
+    }
+
+    /**
+     * Display a list of Media Revisions in the MediaManager
+     *
+     * @param string $id  media id
+     * @return void
+     */
+    protected function showMediaRevisions($id)
+    {
+        global $lang;
+
+        // get revisions, and set correct pagenation parameters (first, hasNext)
+        $first   = $this->first;
+        $hasNext = false;
+        $revisions = $this->getRevisions($first, $hasNext);
+
+        // create the form
+        $form = new Form([
+                'id' => 'page__revisions', // must not be "media__revisions"
+                'action' => media_managerURL(['image' => $id], '&'),
+                'class'  => 'changes',
+        ]);
+        $form->setHiddenField('mediado', 'diff'); // required for media revisions
+        $form->addTagOpen('div')->addClass('no');
+
+        // start listing
+        $form->addTagOpen('ul');
+        foreach ($revisions as $info) {
+            $rev = $info['date'];
+            $class = ($info['type'] === DOKU_CHANGE_TYPE_MINOR_EDIT) ? 'minor' : '';
+            $form->addTagOpen('li')->addClass($class);
+            $form->addTagOpen('div')->addClass('li');
+
+            if (isset($info['current'])) {
+               $form->addCheckbox('rev2[]')->val('current');
+            } elseif (file_exists(mediaFN($id, $rev))) {
+                $form->addCheckbox('rev2[]')->val($rev);
+            } else {
+                $form->addCheckbox('')->val($rev)->attr('disabled','disabled');
+            }
+            $form->addHTML(' ');
+
+            $objRevInfo = $this->getObjRevInfo($info);
+            $html = implode(' ', [
+                $objRevInfo->editDate(),          // edit date and time
+                $objRevInfo->difflink(),          // link to diffview icon
+                $objRevInfo->itemName(),          // name of page or media
+                '<div>',
+                $objRevInfo->editSummary(),       // edit summary
+                $objRevInfo->editor(),            // editor info
+                html_sizechange($info['sizechange']), // size change indicator
```

Comment:
For sizechange of media the `html_sizechange()` is still used. Further it is fine for me to deprecate `html_sizechange()`.